Yona/Yona/Settings/SettingsViewController.swift

```diff
@@ -110,15 +117,21 @@ extension SettingsViewController:UITableViewDelegate {
     }
     
     func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
-        if indexPath.row == 2 {
+        if indexPath.row == 0 {
+            //change pin
+            setViewControllerToDisplay("Passcode", key: YonaConstants.nsUserDefaultsKeys.screenToDisplay)
+            if let passcode = R.storyboard.passcode.passcodeStoryboard {
+                self.navigationController?.pushViewController(passcode, animated: false)
+            }
+        } else if indexPath.row == 1 {
+            //privacy
```

Comment:
What does the different indexPath.row number supposed to mean?
The indexes are meaningless. There are several ways to deal with such thing, but since it seems there is only 3 cases, we can handle them like that
`let setting = settingsArray[indexPath.row]

        if setting == NSLocalizedString("change-pin", comment: ""){
            
        }else if setting ==  NSLocalizedString("privacy", comment: ""){
            
        }else if setting == NSLocalizedString("delete-user", comment: ""){
            
        }`

At least it now contains words instead of numbers, which can help you:
1- Easily guess what this code supposed to do
2- It is not depending on the order any more.